Added ruleset option to the preference page

rakshitkr opened the pull request:
- Automatic download of the rulesets from the nexus server based on the url.
- Update the rulesets on every eclipse restart.
- Integrated with preference page UI and the analysis.

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Added ruleset option to the preference page

- Automatic download of the rulesets from the nexus server based on the url.
- Update the rulesets on every eclipse restart.
- Integrated with preference page UI and the analysis.

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

kruegers commented on plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java:
```diff
@@ -555,11 +556,28 @@ public String toString() {
+	public static final String BC_TABLE_ITEM = "Bouncy Castle";
+	public static final String TINK_TABLE_ITEM = "Tink";
+	
+	public static final String NEXUS_SOOT_RELEASE = "https://soot-build.cs.uni-paderborn.de/nexus/repository/soot-release";
```
Please extract all URLS to an external config file.

rakshitkr replied:
Do you mean a separate java file which consists of only URLs?

kruegers commented on plugins/de.cognicrypt.staticanalyzer/pom.xml:
```diff
@@ -11,6 +10,56 @@
+  
+  <dependencies>
+		<dependency>
+			<groupId>org.eclipse.aether</groupId>
```
Why do you need these dependencies here again? You've already defined them in the POM of core.

rakshitkr replied:
You are right. These redundant dependencies have been removed.

rakshitkr committed:
Removed aether dependencies from staticanalyzer pom AND ruleset dependencies from core pom

Removed the following dependencies aether-api, aether-spi, aether-impl, aether-connector-basic, aether-transport-file, aether-transport-http, maven-aether-provider, zip4j since they are already added in core pom

Signed-off-by: Rakshit Krishnappa Ravi <rakshitkrishnapparavi@gmail.com>

rakshitkr committed:
Merge branch 'develop' of https://github.com/eclipse-cognicrypt/CogniCrypt into eclipse-cognicrypt-develop

rakshitkr committed:
Merge branch 'eclipse-cognicrypt-develop' into add_ruleset_option

Signed-off-by: Rakshit Krishnappa Ravi rakshitkrishnapparavi@gmail.com

rakshitkr committed:
Undid the deletion of code that enables/disables the show secure objects option

Signed-off-by: Rakshit Krishnappa Ravi rakshitkrishnapparavi@gmail.com

sritejakv committed:
Adding support for source files and removing binary option.

- Removed binary parsing of the rules.
- Added library to parse ini files.
- Moved all the urls to configuration.ini file.
- Removing the constant RELATIVE_RULE_DIR.
- Added libraries to the mainfest to support source cryptsl file
parsing.


Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Merge branch 'develop' into add_ruleset_option

sritejakv commented:
Before merging this branch, all the rule versions in the nexus repository should be changed from binary to source format. Now the plugin only supports rules in source format.

kruegers committed:
Merge branch 'develop' into add_ruleset_option

sritejakv committed:
Adding minimum version to the rulesets so that binary rulesets are not
downloaded.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Update ArtifactUtils.java

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Moving rules to resources/CrySLRules/Custom folder.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Correcting the typo of a constant.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Adding UI check box for Custom rules and its preferences. Adding
'Custom' folder inside CrySLRules.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

sritejakv committed:
Merge branch 'develop' into add_ruleset_option

sritejakv commented:
Before merging this branch, please merge [PR-188](https://github.com/CROSSINGTUD/CryptoAnalysis/pull/188) of CryptoAnalysis.

sritejakv commented:
> Before merging this branch, all the rule versions in the nexus repository should be changed from binary to source format. Now the plugin only supports rules in source format.

As a solution for this, minimum rule version is added to skip all the rules in binary format.

sritejakv committed:
Fixing the build error.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

kruegers commented on plugins/de.cognicrypt.core/src/de/cognicrypt/utils/Utils.java:
```diff
@@ -327,6 +334,51 @@ public static File getResourceFromWithin(final String inputPath, final String pl
+		try {
+			ini = new Wini(getResourceFromWithin(Constants.CONFIG_FILE_PATH));
+		} catch (InvalidFileFormatException e) {
+			e.printStackTrace();
```
Please add proper exception handling.

kruegers commented on plugins/de.cognicrypt.core/META-INF/MANIFEST.MF:
```diff
@@ -31,6 +32,7 @@ Bundle-ClassPath: .,
  lib/commons-io.jar,
  lib/commons-lang3.jar,
  lib/CryptoAnalysis.jar,
+ lib/de.darmstadt.tu.crossing.CryptSL.jar,
```
Why have you added this dependency now? This should not be required as all CrySL related classes must be made available through CryptoAnalysis.

kruegers commented on plugins/de.cognicrypt.core/META-INF/MANIFEST.MF:
```diff
@@ -39,14 +41,24 @@ Bundle-ClassPath: .,
+ lib/log4j.jar,
  lib/log4j-api.jar,
  lib/log4j-core.jar,
+ lib/org.eclipse.emf.common.jar,
```
Same question as above for all  org.eclipse dependencies.

kruegers reviewed (changes requested):
Sorry for the delayed review. I hadn't noticed you had addressed all my requests.
Also thanks for this work and for even implementing the customrules option. Please take care of the remaining three comments and the PR is ready to be merged. Thanks.

sritejakv committed:
Adding proper exception handling message.

Signed-off-by: Sriteja Kummita <sriteja.ku@gmail.com>

kruegers committed:
Merge branch 'develop' into add_ruleset_option

Conflicts:
	plugins/de.cognicrypt.core/src/de/cognicrypt/utils/Utils.java
	plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/StaticAnalyzerPreferences.java
	plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/sootbridge/SootRunner.java

kruegers commented:
Hey @rakshitkr,
The branch looks ok to me now, but unfortunately, you seem to have missed to [sign-off some of your commits](https://accounts.eclipse.org/legal/eca/validation/13425). Could you please take care of that? Then, the branch can be merged as is.

kruegers commented:
Closed because replaced by #344 .
